frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/RequestManager.java

```diff
@@ -208,6 +210,33 @@ private void checkFrameworkVersion(FrameworkDescriptor newFrameworkDescriptor) t
     }
   }
 
+  private void checkHadoopLibrary(FrameworkDescriptor newFrameworkDescriptor) throws Exception {
+    if (getFrameworkDescriptorGpuCount(newFrameworkDescriptor) > 0) {
+      if (hadoopLibrarySupportGPU == -1) {
+        if (ResourceDescriptor.checkIfHadoopLibrarySupportGPU()) {
+          hadoopLibrarySupportGPU = 1;
+        } else {
+          hadoopLibrarySupportGPU = 0;
+        }
+      }
+      if (hadoopLibrarySupportGPU == 0) {
+        throw new NonTransientException("this hadoop library doesn't support GPU scheduling, please use the hadoop-AI library");
+      }
+    }
+    if (getFrameworkDescriptorPortCount(newFrameworkDescriptor) > 0) {
+      if (hadoopLibrarySupportPort == -1) {
+        if (ResourceDescriptor.checkIfHadoopLibrarySupportPort()) {
+          hadoopLibrarySupportPort = 1;
+        } else {
+          hadoopLibrarySupportPort = 0;
+        }
+      }
```

Comment:
How about moving this lines to ResourceManager's constructor to init hadoopLibrarySupportPort only once?